packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -721,7 +740,12 @@ export class LocalIntervalCollection<TInterval extends ISerializableInterval> {
     private removeIntervalFromIndex(interval: TInterval) {
         this.intervalTree.removeExisting(interval);
         this.endIntervalTree.remove(interval);
-        this.intervalIdMap.delete(interval.getIntervalId());
+
+        const id = interval.getIntervalId();
+
+        if (id) {
```

Comment:
we should assert here rather than tolerate undefined.

IMO, we should eventually clean up this code to move back-compat of id creation into summary load, then assume all fresh intervals coming over the wire have IDs. I think that's probably out of scope for this change though.